nix/modules/scripts.nix

```diff
@@ -51,7 +58,7 @@
         fi
 
         if [[ "$VERSIONS_DIR" == "$DEFAULT_VERSIONS_DIR" ]]; then
-          nix flake lock --tarball-ttl 0 --update-input versions --override-input versions "path:$VERSIONS_DIR"
+          nix flake update versions
```

Comment:
The old command is giving a deprecation warning and this one does the job neatly.

Reply:
please keep `--tarball-ttl 0` just in case we ever need to run this in burst fashion.

leaving out the override definitely changes the semantics, is that intentional? specifically:
instead of using the versions flake in `$VERSIONS_DIR` it will update according to the flake's specified ref. thus any changes to that flake during this run won't be considered.

i've attempted to tackle this issue a few times in the past and never fully saw it through. the tricky part is that all end-users will not use the `path:..` location, so i prefer the way you've written this. however for it to be correct the workflow needs to push to the branch first and then update against it.

Reply:
`--tarball-ttl 0` is no longer available as an option for the new command.

Yes the input change is intentional. We are never updating the inner `flake.nix` and that's the only thing that it reads here. We update the inner lock but the outer flake update doesn't look at that from what I can tell by looking at the Nix source code. So there's no need to commit or push anything before the second step.

When somebody uses Holonix without an override for `versions` they will get exactly what's on `develop`, currently using the `0_2` version. Doing this update like this here means that the locked `versions` they see will be precisely in sync with what's on `develop`. 

The `if` check I see as a way to ensure that the lock changes resulting from changing the `0_2` version end up on the same PR.